totalRP3/Core/Debug.lua

```diff
@@ -0,0 +1,74 @@
+-- Copyright The Total RP 3 Authors
+-- SPDX-License-Identifier: Apache-2.0
+
+TRP3_DebugUtil = {};
+
+local ObjectDebugNames = setmetatable({}, { __mode = "kv" });
+
+function TRP3_DebugUtil.GetDebugName(object)
+	local name = ObjectDebugNames[object];
+
+	if not name and object.GetDebugName then
+		name = object:GetDebugName();
+	end
+
+	if not name then
+		return tostring(object);
+	end
+
+	return name;
+end
+
+function TRP3_DebugUtil.SetDebugName(object, name)
+	ObjectDebugNames[object] = name;
+end
+
+local EventTracingRegistries = setmetatable({}, { __mode = "kv" });
+
+local function OnCallbackEventFired(registry, event, ...)
+	if not EventTracingRegistries[registry] then
+		return;
+	elseif not EventTrace or not EventTrace.LogLine then
+		return;
+	end
+
+	if TRP3_API.IsEventValid(TRP3_API.GameEvents, event) then
+		return;
+	end
```

Comment:
Will remove this later - this is a leftover from before I added the RemoveFromEventTraceWindow function.

Reply:
Are we still removing this later